src/test/cli/crushtool/reclassify.t

```diff
@@ -0,0 +1,377 @@
+    renumbering bucket -11 -> -142
+    renumbering bucket -32 -> -143
+    renumbering bucket -31 -> -144
+    renumbering bucket -10 -> -145
+    renumbering bucket -8 -> -146
+    renumbering bucket -6 -> -147
+    renumbering bucket -28 -> -148
+    renumbering bucket -27 -> -149
+    renumbering bucket -21 -> -150
+    renumbering bucket -20 -> -151
+    renumbering bucket -19 -> -152
+    renumbering bucket -7 -> -153
+    renumbering bucket -5 -> -154
+    renumbering bucket -4 -> -155
+    renumbering bucket -25 -> -156
+    renumbering bucket -24 -> -157
+    renumbering bucket -23 -> -158
+    renumbering bucket -17 -> -159
+    renumbering bucket -16 -> -160
+    renumbering bucket -15 -> -161
+    renumbering bucket -3 -> -162
+    renumbering bucket -72 -> -163
+    renumbering bucket -98 -> -164
+    renumbering bucket -97 -> -165
+    renumbering bucket -96 -> -166
+    renumbering bucket -95 -> -167
+    renumbering bucket -94 -> -168
+    renumbering bucket -93 -> -169
+    renumbering bucket -68 -> -170
+  classify_root 0513-R-0060 (-65) as hdd
+    new class hdd exists as 0
+    renumbering bucket -65 -> -35
+    renumbering bucket -76 -> -36
+    renumbering bucket -78 -> -37
+    renumbering bucket -87 -> -38
+    renumbering bucket -82 -> -39
+    renumbering bucket -81 -> -40
+    renumbering bucket -77 -> -41
+    renumbering bucket -75 -> -42
+    renumbering bucket -89 -> -43
+    renumbering bucket -85 -> -44
+    renumbering bucket -84 -> -45
+    renumbering bucket -74 -> -46
+    renumbering bucket -71 -> -47
+    renumbering bucket -80 -> -48
+    renumbering bucket -91 -> -49
+    renumbering bucket -90 -> -50
+    renumbering bucket -88 -> -51
+    renumbering bucket -79 -> -52
+    renumbering bucket -70 -> -53
+    renumbering bucket -86 -> -54
+    renumbering bucket -83 -> -55
+    renumbering bucket -73 -> -56
+    renumbering bucket -69 -> -57
+  $ crushtool -i $TESTDIR/crush-classes/beesly --compare foo
+  rule 0 had 0/10240 mismatched mappings (0)
+  rule 1 had 0/10240 mismatched mappings (0)
+  rule 2 had 0/10240 mismatched mappings (0)
+  rule 4 had 0/10240 mismatched mappings (0)
+  maps appear equivalent
+
+  $ crushtool -i $TESTDIR/crush-classes/flax --reclassify --reclassify-root default hdd -o foo
+  classify_root default (-1) as hdd
+    new class hdd exists as 0
+    renumbering bucket -1 -> -5
+    renumbering bucket -12 -> -7
+    renumbering bucket -9 -> -8
+    renumbering bucket -6 -> -10
+    renumbering bucket -4 -> -11
+    renumbering bucket -3 -> -13
+    renumbering bucket -2 -> -14
+  $ crushtool -i $TESTDIR/crush-classes/flax --compare foo
+  rule 0 had 0/10240 mismatched mappings (0)
+  maps appear equivalent
+
+  $ crushtool -i $TESTDIR/crush-classes/gabe --reclassify --reclassify-root default hdd -o foo
+  classify_root default (-1) as hdd
+    new class hdd exists as 0
+    rule 3 includes take on root default class 0
+  failed to reclassify map
+  [1]
```

Comment:
@dvanders Ok, there is somethign else weird going on with this map. I added a few commands to crushtool to help sort it out.. but in the end a simple --reweight on this map makes a huge number of  inputs remap because the weights on the original map are all out of whack.  For example,

```
gnit:build (wip-crush-reclassify) 01:26 PM $ bin/crushtool -i crush-classes/gabe2 --reweight -o reweighted
gnit:build (wip-crush-reclassify) 01:26 PM $ bin/crushtool -i crush-classes/gabe2 --reclassify --reclassify-root default hdd -o reclassified
classify_root default (-1) as hdd
  new class hdd exists as 0
  renumbering bucket -1 -> -178
  renumbering bucket -4 -> -179
  renumbering bucket -25 -> -180
  renumbering bucket -16 -> -181
  renumbering bucket -21 -> -182
  renumbering bucket -19 -> -183
  renumbering bucket -15 -> -184
  renumbering bucket -7 -> -185
  renumbering bucket -47 -> -186
  renumbering bucket -18 -> -187
  renumbering bucket -8 -> -188
  renumbering bucket -6 -> -189
  renumbering bucket -12 -> -190
  renumbering bucket -23 -> -191
  renumbering bucket -22 -> -192
  renumbering bucket -20 -> -193
  renumbering bucket -11 -> -194
  renumbering bucket -10 -> -195
  renumbering bucket -17 -> -196
  renumbering bucket -13 -> -197
  renumbering bucket -9 -> -198
  renumbering bucket -3 -> -199
  renumbering bucket -14 -> -200
  renumbering bucket -5 -> -201
  renumbering bucket -2 -> -202
gnit:build (wip-crush-reclassify) 01:26 PM $ bin/crushtool -i crush-classes/gabe2 --compare reweighted
rule 0 had 581/10240 mismatched mappings (0.0567383)
rule 1 had 368/6144 mismatched mappings (0.0598958)
warning: maps are NOT equivalent
gnit:build (wip-crush-reclassify) 01:26 PM $ bin/crushtool -i crush-classes/gabe2 --compare reclassified
rule 0 had 627/10240 mismatched mappings (0.0612305)
rule 1 had 422/6144 mismatched mappings (0.0686849)
warning: maps are NOT equivalent
```
i.e., we have almost as much data movement just from making the weight-set weights sum properly.  I'm not quite sure why they are out of whack on this map..

But, strangely, if i resum them first, and then reclassify, and compare the before/after, that second reclassify step still moves a little bit of data:
```
$ bin/crushtool -i reweighted --compare both
rule 0 had 90/10240 mismatched mappings (0.00878906)
rule 1 had 78/6144 mismatched mappings (0.0126953)
warning: maps are NOT equivalent
```
...so the handling for the weight-sets isn't perfect.  I'm inclined to say it's close enough for now at least and move on.  This conversion is a one-time thing and moving a little bit is better than moving everything!